src/components/Nav/components/NavAvatar/index.js

```diff
@@ -19,36 +20,64 @@ class GordonNavAvatar extends Component {
     this.getInitials = this.getInitials.bind(this);
 
     this.state = {
-      user: {
-        name: 'Test User',
-        email: 'test.user@gordon.edu',
-        username: 'test.user',
-      },
+      email: null,
+      image: null,
+      name: null,
+      username: null,
     };
   }
+  async componentWillMount() {
+    const { name, user_name: username } = user.getLocalInfo();
+    this.setState({ name, username });
+    const [
+      { Email: email },
+      { def: defaultImage, pref: preferredImage },
+    ] = await Promise.all([
+      await user.getProfileInfo(),
+      await user.getImage(),
+    ]);
+
+    const image = preferredImage || defaultImage;
```

Comment:
So this is the first time I'm seeing this notation. I get the idea of it but how does it decide which one to do, or is it logically have to be one or the other?

Reply:
Good question! `||` means "or", and JavaScript uses [falsy/truthy checking](https://www.sitepoint.com/javascript-truthy-falsy/), so this line essentially means:
"the variable `image` should equal `preferredImage` if it is truthy or `defaultImage` otherwise".

When `preferredImage` is defined, it will be truthy, so it will be used as the value of `image`, and when it is undefined, it will be falsy, so `defaultImage` will be used as the value of `image` instead.